refresh the documentation of the plugin adoption process

jmMeessen opened the pull request:
The existing documentation became a little dated and did not reflect the current practice. This is a proposal for a documentation update (unless the current adoption practice should be ironed out).

MarkEWaite commented on content/doc/developer/plugin-governance/adopt-a-plugin.adoc:
```diff
@@ -20,23 +20,44 @@ This page addresses cases when there is no active maintainer of a plugin:
+
+The procedure is two-fold: 
+
+==== announce your adoption intention by mail 
```
General guidance is to use sentence case, so start with an upper case letter
```suggestion
==== Announce your adoption intention by mail 
```

MarkEWaite commented on content/doc/developer/plugin-governance/adopt-a-plugin.adoc:
```diff
@@ -20,23 +20,44 @@ This page addresses cases when there is no active maintainer of a plugin:
+
+==== announce your adoption intention by mail 
+
+Send the notification email to the http://jenkins-ci.org/content/mailing-lists[Jenkins Developers mailing list].
```
Better to link directly to the mailing list
```suggestion
Send the notification email to the https://groups.google.com/g/jenkinsci-dev[Jenkins Developers mailing list].
```

MarkEWaite replied:
If you prefer to remain with the link to the lists page, please switch the link to `link:/mailing-lists/` so that it will work for local site development and will use HTTPS immediately instead of redirecting through HTTP to HTTPS

MarkEWaite commented on content/doc/developer/plugin-governance/adopt-a-plugin.adoc:
```diff
@@ -20,23 +20,44 @@ This page addresses cases when there is no active maintainer of a plugin:
+Be sure to provide the following information: 
 
 * Link to a plugin you want to adopt
+* The status of the plugin ("for adoption" or "abandonned")
```
Spelling fix
```suggestion
* The status of the plugin ("for adoption" or "abandoned")
```

MarkEWaite commented on content/doc/developer/plugin-governance/adopt-a-plugin.adoc:
```diff
@@ -20,23 +20,44 @@ This page addresses cases when there is no active maintainer of a plugin:
 * Your Jenkins infrastructure account id. link:https://accounts.jenkins.io/[Create your account] if you don't have one.
+* The link to the "Repository Permission Updater" PR described below
+
+==== submit a PR on the "Repository Permission Updater" (RPU) repository
```
Sentence case and probably don't need the acronym "RPU" in the title
```suggestion
==== Submit a PR on the "Repository Permission Updater" repository
```

MarkEWaite commented on content/doc/developer/plugin-governance/adopt-a-plugin.adoc:
```diff
@@ -20,23 +20,44 @@ This page addresses cases when there is no active maintainer of a plugin:
+Add your Jenkins infrastructure account under the `developer` section.
+Refer to the link:https://github.com/jenkins-infra/repository-permissions-updater/blob/master/README.md[Readme] for detailed guidance.
+
+It is a good practice to ping the current registered developers on the PR (just in case they miss the notification pn the Jenkins Dev List).
```
```suggestion
It is a good practice to ping the current registered developers on the PR (just in case they miss the notification on the Jenkins Dev List).
```

MarkEWaite commented on content/doc/developer/plugin-governance/adopt-a-plugin.adoc:
```diff
@@ -20,23 +20,44 @@ This page addresses cases when there is no active maintainer of a plugin:
+It is a good practice to ping the current registered developers on the PR (just in case they miss the notification pn the Jenkins Dev List).
+
+Once you got access to the plugin's Github repository, don't forget to remove the "adopt-this-plugin" topic.
+This can be done by clicking on the gear wheel near the about.
```
Since we're trying to point the user to an exact text location on the page, I think it might help if the precise case of the text is matched.
```suggestion
This can be done by clicking on the gear wheel near the "About".
```

MarkEWaite commented on content/doc/developer/plugin-governance/adopt-a-plugin.adoc:
```diff
@@ -20,23 +20,44 @@ This page addresses cases when there is no active maintainer of a plugin:
+Once you got access to the plugin's Github repository, don't forget to remove the "adopt-this-plugin" topic.
+This can be done by clicking on the gear wheel near the about.
+
+=== Abandonned plugins
```
Spelling fix
```suggestion
=== Abandoned plugins
```

MarkEWaite reviewed (approved):
Optional comments.  Will need review and approval from at least one other member of the governance board.

@halkeye , @ewelinawilkosz , @oleg-nenashev , @kohsuke

halkeye commented:
Has there been any agreed upon change to the adoption process? I think mark and I have just been telling people to make a PR out of (I'm sorry mark) lazyness and organization.  It may want to have a quick shout on the mailing list before any sort of board approval.

Also it reminded me that I still have too many projects on the go. I want to get https://plugins-self-service-3ir4b.ondigitalocean.app/ deployed onto jenkins.io infra to make it easier to setup labels and stuff for new plugins.

I'll try to give it a read over once work is done and sign off on it.

MarkEWaite commented:
> Has there been any agreed upon change to the adoption process? I think mark and I have just been telling people to make a PR out of (I'm sorry mark) lazyness and organization. It may want to have a quick shout on the mailing list before any sort of board approval.

I don't think this is trying to describe any concrete changes in the adoption process.  It is trying to describe current practices more clearly.  If there is a significant change due to these instructions, I believe that is unintentional.

jmMeessen commented:
As @MarkEWaite mentioned, this PR does not propose changes to process but rather match the documentation with the current practice. If the current practice doesn't match the process, we should make the practice clearer with formal sign off in the mail part for example. And still clarify the documentation.
It is an important process in our community lifecycle.

It is because I had doubts that I kept this PR as a draft.

halkeye commented:
I'm sorry I didn't get a chance to review this after work, I fell asleep, I'll try to do it tomorrow

jmMeessen commented:
@halkeye Did you have a chance to read and review this.

timja commented on content/doc/developer/plugin-governance/adopt-a-plugin.adoc:
```diff
@@ -18,25 +18,44 @@ This page addresses cases when there is no active maintainer of a plugin:
-Once granted access, you can file a pull request against link:https://github.com/jenkins-infra/repository-permissions-updater[Repository Permission updater] to be able to deploy snapshots and releases for your plugin.
-You're generally expected to start slowly, by filing PRs, and not commit directly.
-Even more for plugins with a big number of installations for obvious reasons.
+Once the situation clarified, proceed as for a "adoptable" plugins (see above). 
```
```suggestion
Once the situation is clarified, proceed as for a "adoptable" plugins (see above). 
```

timja commented on content/doc/developer/plugin-governance/adopt-a-plugin.adoc:
```diff
@@ -18,25 +18,44 @@ This page addresses cases when there is no active maintainer of a plugin:
-You're generally expected to start slowly, by filing PRs, and not commit directly.
-Even more for plugins with a big number of installations for obvious reasons.
+Once the situation clarified, proceed as for a "adoptable" plugins (see above). 
+On the notification email, add information about the steps taken to contact the exisiting maintainers.
```
```suggestion
On the notification email, add information about the steps taken to contact the existing maintainers.
```

timja reviewed (approved):
Looks fine to me, it doesn't change any processes just helps clarify from what I can tell

halkeye commented:
I'm glad tim had a chance to review. I will absolutely get to it today. This weekend ended up being mostly chores and me time.

halkeye commented on content/doc/developer/plugin-governance/adopt-a-plugin.adoc:
```diff
@@ -18,25 +18,44 @@ This page addresses cases when there is no active maintainer of a plugin:
+
+**Submit a PR on the "Repository Permission Updater" repository**
+
+File a pull request against link:https://github.com/jenkins-infra/repository-permissions-updater[Repository Permission updater] to be able to deploy snapshots and releases for your plugin.
```
I don't have a strong opinion but I kind feel like there's not much point in mentioning snapshots, and I think it confuses people when they try and it doesn't show up in any release center?

jmMeessen replied:
Good point. Will correct accordingly.

halkeye commented on content/doc/developer/plugin-governance/adopt-a-plugin.adoc:
```diff
@@ -18,25 +18,44 @@ This page addresses cases when there is no active maintainer of a plugin:
+Add your Jenkins infrastructure account under the `developer` section.
+Refer to the link:https://github.com/jenkins-infra/repository-permissions-updater/blob/master/README.md[Readme] for detailed guidance.
+
+It is a good practice to ping the current registered developers on the PR (just in case they miss the notification on the Jenkins Dev List).
```
I would make this more mandatory, good practice says to me "You should do it, but okay if it isn't". We as hosting team want approval from the existing maintainer, the two week countdown is after the attempt is made.

timja replied:
Isn't this section about adopting abandoning plugins where the timeout occurred in the mailing list?

halkeye replied:
Ah yea, I guess for up for adoption it's optional, for timeout ones it's mandatory? I would just say "Please ping" instead of "Its good practice" incase the existing maintainer wants to knowledge transfer or be informed or whatever. But yea, way less mandatory for up for adoption plugins

jmMeessen replied:
Good suggestion.

halkeye reviewed (approved):
Couple of minor nits, but overall I like the improvements, signing off as board and hosting team member

halkeye reviewed (approved):
Couple of minor nits, but overall I like the improvements, signing off as board and hosting team member

halkeye commented:
looking at https://deploy-preview-4820--jenkins-io-site-pr.netlify.app/doc/developer/plugin-governance/adopt-a-plugin/ makes me so happy :D :D :D

Might also, in this one, or a follow up, be worth an FAQ item about removing the security banners. It comes up a bit for new maintainers.

timja commented on content/doc/developer/plugin-governance/adopt-a-plugin.adoc:
```diff
@@ -18,25 +18,44 @@ This page addresses cases when there is no active maintainer of a plugin:
+Once you got access to the plugin's Github repository, don't forget to remove the "adopt-this-plugin" topic.
+This can be done by clicking on the gear wheel near the "About".
+
+== Abandonned plugins
```
```suggestion
== Abandoned plugins
```

jmMeessen replied:
Looking for an asciidoc spellcheck solution (preferably in VSC)... That would help non-native english speakers

timja replied:
this works fine for me: https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker

jmMeessen commented:
Chiming in with @halkeye , the preview feature is indeed a great addition and improves the non-technical reviewers experience.

StackScribe commented on content/doc/developer/plugin-governance/adopt-a-plugin.adoc:
```diff
@@ -18,25 +18,44 @@ This page addresses cases when there is no active maintainer of a plugin:
+
+Ping the current registered developers on the PR to inform them of your initiative (just in case they miss the notification on the Jenkins Dev List).
+
+Once you got access to the plugin's Github repository, don't forget to remove the "adopt-this-plugin" topic.
```
```suggestion
Once you have access to the plugin's Github repository, be sure to remove the "adopt-this-plugin" topic.
```

MarkEWaite commented:
@jmMeessen was this closed intentionally?  It seemed like the instructions were an improvement.

jmMeessen commented:
Oops. I cleaned my repo and deleted the banch. Thought it was merged.

On Fri, 21 Jan 2022, 20:08 Mark Waite, ***@***.***> wrote:

> @jmMeessen <https://github.com/jmMeessen> was this closed intentionally?
> It seemed like the instructions were an improvement.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jenkins-infra/jenkins.io/pull/4820#issuecomment-1018784143>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABN3S7NIN2E3AAD56BPPRXLUXGVMFANCNFSM5L46TJ6Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

halkeye commented:
Enable auto merge and merge it!

Any more follow-ups can come later it's still a win
